### PR TITLE
[NEW] Livechat

### PR DESCRIPTION
**Pull request by djorkaeffalexandre**
@RocketChat/ReactNative

Now we'll support all livechat actions and show all livechat data/info on mobile client.

- [X] Close
- [X] Forward
- [X] Return
- [x] Edit
- [x] Show navigation actions

- [x] Check `you was removed from room`.
- [ ] Add new icons.

Screen Shot 2020-04-13 at 15 01 41

Screen Shot 2020-04-07 at 12 50 32

Screen Shot 2020-04-09 at 15 16 37

Screen Shot 2020-04-07 at 14 08 33

Screen Shot 2020-04-07 at 14 14 28

Screen Shot 2020-04-08 at 10 26 31

Screen Shot 2020-04-08 at 17 49 46

### CODE REVIEW

File: app/views/ForwardLivechatView.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+const ForwardLivechatView = ({ navigation, theme }) => {
+	const [departments, setDepartments] = useState([]);
+	const [departmentId, setDepartment] = useState();
+	const [users, setUsers] = useState([]);
+	const [userId, setUser] = useState();
+	const [room, setRoom] = useState();
+
+	const rid = navigation.getParam('rid');
+
+	const getDepartments = async() => {
```
Comment (djorkaeffalexandre) [inline app/views/ForwardLivechatView.js:31]: Request these things when open the page instead of here.

File: app/views/ForwardLivechatView.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+			title,
+			data: users,
+			onChangeValue: setUser,
+			onChangeText: getUsers
+		});
+	};
+
+	return (
+		<View style={[styles.container, { backgroundColor: themes[theme].auxiliaryBackground }]}>
+			<ScrollView style={styles.content}>
```
Comment (djorkaeffalexandre) [inline app/views/ForwardLivechatView.js:131]: I think it's not needed.

File: app/views/ForwardLivechatView.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+					title={I18n.t('Forward_to_department')}
+					onPress={title => onPressDepartment(title)}
+					theme={theme}
+				/>
+				<OnboardingSeparator theme={theme} />
+				<ListItem
+					title={I18n.t('Forward_to_user')}
+					onPress={title => onPressUser(title)}
+					theme={theme}
+				/>
```
Comment (djorkaeffalexandre) [inline app/views/ForwardLivechatView.js:142]: Use some that looks like an input instead of these.

File: app/views/ForwardLivechatView.js (hunk 1 above)
Comment (djorkaeffalexandre) [reply app/views/ForwardLivechatView.js:31]: Is better if we request these things here.

File: app/views/RoomInfoView/Livechat.js [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+	const getRoom = () => {
+		if (room.departmentId) {
+			getDepartment(room.departmentId);
+		}
+	};
+
+	useEffect(() => { getRoom(); }, []);
+
+	return (
```
Comment (diegolmello) [inline app/views/RoomInfoView/Livechat.js:49]: Can we show `username` on this screen?

File: app/lib/rocketchat.js [hunk 5]
```diff
@@ -824,6 +824,46 @@ const RocketChat = {
 		return this.sdk.get('rooms.info', { roomId });
 	},
 
+	getVisitorInfo(visitorId) {
```
[the remark this answers is not in the document]
Comment (djorkaeffalexandre) [reply app/lib/rocketchat.js:752]: Unfortunately I can't find these methodCalls versions, it's not documented and some are commits from 4 years ago, so it's not documented on GitHub release tags. 😢

**Pull request merged**